packages/ckeditor5-dev-release-tools/lib/utils/executeinparallel.js

```diff
@@ -26,65 +25,80 @@ const WORKER_SCRIPT = upath.join( __dirname, 'parallelworker.js' );
  * @see https://nodejs.org/api/worker_threads.html
  * @param {Object} options
  * @param {String} options.packagesDirectory Relative path to a location of packages to execute a task.
- * @param {String} options.processDescription Description of the task displayed on a screen.
  * @param {Function} options.taskToExecute A callback that is executed on all found packages.
  * It receives an absolute path to a package as an argument. It can be synchronous or may return a promise.
  * @param {AbortSignal} options.signal Signal to abort the asynchronous process.
+ * @param {ListrTaskObject} options.listrTask An instance of `ListrTask`.
  * @param {String} [options.cwd=process.cwd()] Current working directory from which all paths will be resolved.
  * @param {Number} [options.concurrency=require( 'os' ).cpus().length / 2] Number of CPUs that will execute the task.
  * @returns {Promise}
  */
-module.exports = function executeInParallel( options ) {
+module.exports = async function executeInParallel( options ) {
 	const {
 		packagesDirectory,
-		processDescription,
 		signal,
 		taskToExecute,
+		listrTask,
 		cwd = process.cwd(),
 		concurrency = require( 'os' ).cpus().length / 2
 	} = options;
 
 	const normalizedCwd = upath.toUnix( cwd );
-	const packages = globSync( `${ packagesDirectory }/*/`, {
+	const packages = await glob( `${ packagesDirectory }/*/`, {
 		cwd: normalizedCwd,
 		absolute: true
 	} );
 	const packagesInThreads = getPackagesGroupedByThreads( packages, concurrency );
 
 	const callbackModule = upath.join( cwd, crypto.randomUUID() + '.js' );
-	fs.writeFileSync( callbackModule, `'use strict';\nmodule.exports = ${ taskToExecute };`, 'utf-8' );
+	await fs.writeFile( callbackModule, `'use strict';\nmodule.exports = ${ taskToExecute };`, 'utf-8' );
 
-	const counter = tools.createSpinner( processDescription, { total: packages.length } );
-	counter.start();
+	const onPackageDone = progressFactory( listrTask, packages.length );
```

Comment:
Looks like `progressFactory` could be passed as an argument to `executeInParallel`. This way `executeInParallel` would be independent of the task runner if we chose to use a different one in the future for other use cases.

Reply:
I agree, but it's hard to predict what arguments it should receive. I would keep it in mind in the future to refactor it.